File: T6_Herencia_Trabajadores/src/model/Jefe.java
package model;
import controller.Empresa;

public class Jefe extends Trabajador{

    private int acciones;
    private double beneficio;

    public Jefe() {}

    public Jefe(String nombre, String apellido, String dni, int acciones, double beneficio) {
        super(nombre, apellido, dni);
        this.acciones = acciones;
        this.beneficio = beneficio;
    }

    public void mostarDatos() {
        super.mostrarDatos();
        System.out.println("Acciones: "+acciones);
        System.out.println("Beneficio: "+beneficio);
    }

    public void despedirTrabajador(String dni) {

    }
}

File: DAM2_Pro/PMDM/Tema_1/TrabajoDeHerencia/src/model/Jefe.kt
package model

import kotlin.random.Random
import kotlin.random.nextInt

class Jefe (nombre: String, apellido: String, dni: String, salario: Double, var nivelResponsabilidad: Int)
    : Trabajador(nombre,apellido,dni,salario) {


    override fun calcularSalarioNeto(): Double {
        nivelResponsabilidad = Random.nextInt(1,6)
        if (nivelResponsabilidad >= 3) {
            salario *= 1.03
            return salario
        } else {
            salario *= 0.97
            return salario
        }
    }


    override fun mostrarDatos() {
        super.mostrarDatos()
        println("Nivel responsabilidad: $nivelResponsabilidad")
    }

}